File: Core/Entities/Message.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Core.Entities
{
    public class Message : BaseEntity
    {
        [Required]
        [MaxLength(255)]
        public string FirstLastName { get; set; }

        [Required]
        [MaxLength(100)]
        public string Email { get; set; }

        [Required]
        [MaxLength(100)]
        public string Phone { get; set; }

        [Required]
        [MaxLength(2000)]
        public string MessageContent { get; set; }
        public bool IsReplied { get; set; }

        [DataType(DataType.Date)]
        public DateTime SendingDate { get; set; } = DateTime.UtcNow;

    }
}

File: angular-happykids/src/app/shared/models/message.ts
export interface ClientMessage {
    id: number;
    firstLastName: string;
    email: string;
    phone: string;
    messageContent: string;
    isReplied: boolean;
    sendingDate: Date;
}

export interface MessageCreate {
    firstLastName: string;
    email: string;
    phone: string;
    messageContent: string;
}

export class MessageEdit {
    id: number;
    firstLastName: string;
    email: string;
    phone: string;
    messageContent: string;
    sendingDate: Date;
    isReplied: boolean;
}
